Feature branch/rails forms

bini-i opened the pull request:
## This is a rails form project from microverse rails curriculum with features of 
- Creating, 
- Reading, 
- Updating, and 
- Deleting a user.

bini-i committed:
Add User model

- with attributes of username, email and password

bini-i committed:
Add validation to User model in user.rb

- validates presence of all attributes

tnyandoro committed:
Add The user controller

Create new resources

 Create new view

bini-i committed:
Add create action in users_controller.rb and new.html.erb view

- create action creates and saves a user object to the database
- view displays a form to create a user

tnyandoro committed:
Add parmas

bini-i committed:
Update new.html.erb view

- substitute the html form with a form created by form_tag helper method

bini-i committed:
Update users_controller.rb

- update create action to create the user object with out whitelisting

tnyandoro committed:
modify the #new action and store it in an instance variable called user

 Coment out #form tag in the app view

 Rebuild the form using #form_for

Play with the input  methods

bini-i committed:
Add update and edit actions(methods) in users_controller.rb

- edit action renders edit view with a form_for form to update the user
- update action updates user object

bini-i committed:
Update update action in users_controller.rb

- add @user.update to update of user objects

tnyandoro committed:
Add Error messages

bini-i committed:
Add Delete action in users_controller.rb and README.md

- looks for a user record and deletes it from the database

bini-i committed:
Add git workflow

- added github/workflows/linters.yml and .rubocop.yml

bini-i committed:
Add git workflow

- added github/workflows/linters.yml and .rubocop.yml

voscarmv commented on app/controllers/users_controller.rb:
```diff
@@ -0,0 +1,47 @@
+    end
+  end
+
+  def update
+    @user = User.find(params[:id])
+    @user.update(user_params)
+    @user.save
+    redirect_to user_path(@user.id)
+  end
```
Please make sure to `redirect_to` **only** if `@user.save` was successful. Otherwise, just `render :edit`. You may use an `if` statement for this.

voscarmv commented on app/views/users/new.html.erb:
```diff
@@ -0,0 +1,75 @@
+<%# <form method="POST" action="/users" accept-charset="UTF-8"> %>
```
Your commented form here should use the `users_path` helper function, thus:

```
`<form action="users_path" method="post" accept-charset="UTF-8">`
```

voscarmv reviewed (changes requested):
# Impressive work, team 💻 👀 ✨ 

- Most of the project requirements have been met.

## However, there are some changes required 👀 🧰 🔨 🔧 

- Please only use the `:new, :create, :edit, :update` routes as specified by Odin: 
![image](https://user-images.githubusercontent.com/2739245/101048925-a78ec080-3548-11eb-908c-9f8c7508ddf3.png)
- Please check my code comments for details on fixing your project!


### Submit these changes and request a new code review! 😄 👍 

### Questions or comments? Reach me on [Twitter](https://twitter.com/voscarmv) or [LinkedIn](https://linkedin.com/in/oscar-mier) 🧑‍💻🗨️🧑‍💻 
# Happy Hacking! 🧠 ⌨️ 🖱️ 🖥️

bini-i committed:
Update update action in users_controller.rb and removed index and show
routes from routes.rb

- update action made to redirect only if save is successful

teekaytech commented on app/controllers/users_controller.rb:
```diff
@@ -0,0 +1,50 @@
+    # @user = User.new(username: params[:username], email: params[:email], password: params[:password])
+
+    if @user.save
+      redirect_to user_path(@user.id)
```
- [ ] `redirect_to new_user_path` if user was saved ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/forms/user_create.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/forms))

teekaytech commented on app/controllers/users_controller.rb:
```diff
@@ -0,0 +1,50 @@
+    @user = User.find(params[:id])
+    @user.update(user_params)
+    if @user.save
+      redirect_to user_path(@user.id)
```
- [ ] Use `redirect_to edit_user_path` if user was saved

teekaytech reviewed (changes requested):
## Status: Changes Required :warning: :vertical_traffic_light: 

### Hello team :wave:

__You've made laudable progress in this project. You've also done a great job in fixing the required changes by the previous reviewer.
However, some changes are still required before this project can be approved. Please check my comments below and fix them as appropriate.__

**[ REQUIRED CHANGES ]**

- [x] I added some inline comments to your code. Kindly check and fix them as appropriate.

- [x] As mentioned by the previous reviewer, your commented form here should use the users_path helper function, thus:

`<form action="users_path" method="post" accept-charset="UTF-8">`



* * *
**Please make the required changes and submit another review request when you are done.**

* * *

**Taofeek Olalere
[Twitter](https://twitter.com/ola_lere) | [Github](https://github.com/teekaytech) | @taofeekolalere on Slack**
__Happy Coding :smiley:__

tnyandoro committed:
merge files

tnyandoro committed:
add requested changes user_path and the helper method

peterrobert commented on app/views/users/new.html.erb:
```diff
@@ -0,0 +1,75 @@
+        <td>
+            <%= f.label :email %>
+        </td>
+        <td>
+            <%= f.text_field :email, placeholder:" email@example.com" %>
+        </td>
```
This field should not be a text_field. It should be an email_field. This is important because it makes sure that what is added in the field is only an email address.

peterrobert commented on app/views/users/edit.html.erb:
```diff
@@ -0,0 +1,42 @@
+    <%= f.label :username, 'Your username here' %> 
+    <%= f.text_field :username %>
+
+    <%= f.label :email %>
+    <%= f.text_field :email, placeholder:" email@example.com" %>
+
```
Kindly make sure that this is an email_field

peterrobert reviewed (changes requested):
### CHANGES REQUIRED :recycle: :100: 
HELLO @bini-i :wave: VERY NICE PROJECT. I LOVE THAT YOU HAVE WORKED ON ALL THE CHANGES SUGGESTED BY THE PREVIOUS TSE. HOWEVER, THERE IS ONE MORE CHANGE THAT IS REQUIRED BEFORE YOU CAN MOVE ON KINDLY CHECK BELOW :arrow_down: 

### [OPTIONAL]

- [ ] You can create a partial for the user form and use it for both the edit and the new HTML views.

if you would like to learn more about partials kindly check [here](https://guides.rubyonrails.org/layouts_and_rendering.html);


### if you have any questions kindly reach me on slack @peterrobert

### HAPPY CODING :computer:

tnyandoro committed:
add email field

peterrobert reviewed (approved):
### ###  APPROVED :+1: :100: 

VERY NICE WORK. GOOD JOB IMPLEMENTING ALL THE REQUIRED CHANGES.

ON TO THE NEXT PROJECT.

### HAPPY CODING :computer:
